crates/ty_python_semantic/src/types/infer/builder.rs

```diff
@@ -1530,7 +1534,7 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
         }
         // In the following cases, the bound type may not be the same as the RHS value type.
         if let AnyNodeRef::ExprAttribute(ast::ExprAttribute { value, attr, .. }) = node {
-            let value_ty = self.try_expression_type(value).unwrap_or_else(|| {
+            let value_ty = self.try_expression_type(value.as_ref()).unwrap_or_else(|| {
```

Comment:
We could probably implement `Into<ExpressionNodeKey>` for `&Box<Expr>` to avoid the need for this?